### PR TITLE
Document the occ encryption:hsmdaemon command

### LINKED ISSUE #2112: Document encryption:hsmdaemon occ commands
## What Needs to be Documented?

The occ commands related to the hsmdaemon are not documented. See `occ encryption:hsmdaemon --help`.

## Where Does This Need To Be Documented?

- https://doc.owncloud.com/server/admin_manual/configuration/server/occ_command.html#encryption
- https://doc.owncloud.com/server/admin_manual/configuration/server/security/hsmdaemon/

## Why Should This Change Be Made? (Optional)

Document available/supported functionality.

## What Type Of Content Change Is This? (Optional)
- [x] New Content Addition
- [ ] Old Content Deprecation
- [ ] Existing Content Simplification
- [ ] Bug Fix to Existing Content

## Which Manual Does This Relate To? (Optional)
- [x] Admin Manual
- [ ] Developer Manual
- [ ] User Manual
- [ ] Android
- [ ] iOS
- [ ] Branded Clients
- [ ] Desktop Client
- [ ] Other

### PR DESCRIPTION
**Pull request by settermjd**
This fixes #2112.

### CODE REVIEW

**Comment by enbrnz** [comment]
Sorry, but I don't know how I would be able to fact check the statements of this new documentation.
Perhaps @pako81 knows more?

**Comment by settermjd** [comment]
> Sorry, but I don't know how I would be able to fact check the statements of this new documentation.
> Perhaps @pako81 knows more?

np. Thanks for saying so. @pako81?

**Comment by pako81** [comment]
@settermjd to less knowledge on this, sorry. maybe asking @sharidas ?

**Comment by settermjd** [comment]
> @settermjd to less knowledge on this, sorry. maybe asking @sharidas ?

shall do.

File: modules/admin_manual/pages/configuration/server/occ_command.adoc [hunk 1]
```diff
@@ -616,6 +619,32 @@ Given the size of your ownCloud filesystem, this may take some time to complete.
...
+----
+{occ-command-example-prefix} encryption:hsmdaemon [options]
+----
+
+===== Options
+
+[width="100%",cols="40%,60%",]
+|===
+| `--export-masterkey` 
+| Export the private master key, base64 encoded.
```
Comment (pmaier1) [inline modules/admin_manual/pages/configuration/server/occ_command.adoc]: This is too little information yet. 
- Does it export the private master key from the HSM? Is this even possible?
- Is it another private master key in the instance? 
- Do I have to worry about security?

Comment (settermjd) [reply modules/admin_manual/pages/configuration/server/occ_command.adoc]: Thanks for the feedback.

Comment (sharidas) [reply modules/admin_manual/pages/configuration/server/occ_command.adoc]: No it does not export the master key from hsm. Its the master key from ownCloud's instance getting exported. To be precise the file residing in `data/files_encryption/OC_DEFAULT_MODULE/master_*.privateKey`.

**Comment by settermjd** [comment]
@sharidas, can you review again, please?

File: modules/admin_manual/pages/configuration/server/occ_command.adoc [hunk 2]
```diff
@@ -618,6 +621,45 @@ Given the size of your ownCloud filesystem, this may take some time to complete.
...
+----
+{occ-command-example-prefix} encryption:hsmdaemon [options]
+----
+
+===== Options
+
+[width="100%",cols="40%,60%",]
+|===
+| `--export-masterkey` 
+a| This command exports a base64-encoded representation of an ownCloud server’s master key private key file, which is used by its default encryption module.
```
Comment (sharidas) [inline modules/admin_manual/pages/configuration/server/occ_command.adoc]: The master private key has the key ID which is there in the hsm device. May be there is a chance that user might get confused. Mind having a sentence which does not imply that user need not worry about exporting the data in the private key as it has only the id?

Comment (settermjd) [reply modules/admin_manual/pages/configuration/server/occ_command.adoc]: So the user will never need to use this command?

Comment (settermjd) [reply modules/admin_manual/pages/configuration/server/occ_command.adoc]: Or, is this command only meant for certain types of users?

Comment (sharidas) [reply modules/admin_manual/pages/configuration/server/occ_command.adoc]: > is this command only meant for certain types of users?

I hope you are not getting confused with user-keys encryption. All I mean to say is this command would give the base64-encoded representation of the private key. And this private key (i.e, the one in the ownCloud instance and not in the hsm) has the id of the private key in th hsm device. Let me know if you still need clarification.

Comment (settermjd) [reply modules/admin_manual/pages/configuration/server/occ_command.adoc]: Sorry for the slow delay on this. I'll get in touch if I need further information.

Comment (settermjd) [reply modules/admin_manual/pages/configuration/server/occ_command.adoc]: @sharidas, this has been updated in ab86f07. I hope it's now correct. Please let me know if it is not.

Comment (settermjd) [reply modules/admin_manual/pages/configuration/server/occ_command.adoc]: @sharidas, any further feedback?

**Comment by mmattel** [comment]
backport to 10.3 necessary

**Pull request merged**